packages/builtin/string.pony

```diff
@@ -21,24 +21,31 @@ class val String is (Seq[U8] & Comparable[String box] & Stringable)
     _ptr = Pointer[U8]._alloc(_alloc)
     _set(0, 0)
 
-  new val from_array(data: Array[U8] val) =>
+  new val from_array(data: Array[U8] val, offset: USize = 0) =>
```

Comment:
`offset` in the context of `String` class means index which can also be negative (meaning index from the right side). For consistency this parameter could be renamed or the code could support negative indices.